### ISSUE TITLE
no-var false positive in TS declare { var ... } block

### ISSUE THREAD

**Issue by mdesantis** [denoland/deno_lint#1363]
### Lint Name

`no-var`

### Code Snippet

```ts
declare global {
  namespace globalThis {
    var test: string
  }
}
```

### Expected Result

No lint warning, as `declare { var ... }` is fine AFAIK

### Actual Result

    `var` keyword is not allowed

### Additional Info

Somehow related to #529 I guess

### Version

```
deno 2.1.1 (stable, debug, x86_64-unknown-linux-gnu)
v8 13.0.245.12-rusty
typescript 5.6.2
```
